src/install.ts

```diff
@@ -1,197 +1,269 @@
+      channel.appendLine(msg);
+      window.showErrorMessage(msg);
+      return null;
+    }
+    return binPath;
+  }
 
+  // Check for the latest release in Github
+  const releaseUrl = `https://api.github.com/repos/${releaseRepository}/releases/latest`;
+  channel.appendLine(
+    `Auto-update is enabled. Fetching latest release from ${releaseUrl}`,
+  );
 
-    // Check for the latest release in Github
-    const releaseUrl = `https://api.github.com/repos/${releaseRepository}/releases/latest`;
-    channel.appendLine(`Auto-update is enabled. Fetching latest release from ${releaseUrl}`);
+  let releaseData: { name?: string } = {};
+  let latestVersion = '';
+  try {
+    const body = await githubApiRequest(releaseUrl);
+    releaseData = JSON.parse(body);
+    latestVersion = releaseData.name;
+    if (latestVersion.startsWith('v')) {
+      latestVersion = latestVersion.substring(1);
+    }
+  } catch (e) {
+    const msg = `Failed to fetch latest release from ${releaseUrl}`;
+    channel.appendLine(msg);
+    channel.appendLine(e);
 
-    let releaseData: { name?: string } = {};
-    let latestVersion = "";
-    try {
-        const body = await githubApiRequest(releaseUrl);
-        releaseData = JSON.parse(body);
-        latestVersion = releaseData.name;
-        if (latestVersion.startsWith("v")) {
-            latestVersion = latestVersion.substring(1);
-        }
-    } catch (e) {
-        const msg = `Failed to fetch latest release from ${releaseUrl}`;
-        channel.appendLine(msg);
-        channel.appendLine(e);
+    if (!isCustomBinPath) {
+      window.showErrorMessage(msg);
+      throw new Error(msg);
+    }
 
-        if (!isCustomBinPath) {
-            window.showErrorMessage(msg);
-            throw new Error(msg);
-        }
+    window.showWarningMessage(
+      msg + '. Continuing with the current version.',
+    );
+    return binPath;
+  }
+  channel.appendLine(`Latest release is ${latestVersion}`);
 
-        window.showWarningMessage(msg + ". Continuing with the current version.");
-        return binPath;
+  // Check the current version
+  let doUpdate = false;
+  if (!binPathExists) {
+    // The binary does not exist. Only install if the user says yes.
+    const value = await window.showInformationMessage(
+      `The jsonnet ${displayName} does not seem to be installed. Do you wish to install the latest version?`,
+      'Yes',
+      'No',
+    );
+    if (value === 'No') {
+      return null;
     }
-    channel.appendLine(`Latest release is ${latestVersion}`);
+    doUpdate = true;
+  } else {
+    // The binary exists
+    try {
+      // Check the version
+      let currentVersion = '';
+      const result = execFileSync(binPath, ['--version']);
+      const prefix = `${binaryName} version `;
+      if (result.toString().startsWith(prefix)) {
+        currentVersion = result
+          .toString()
+          .substring(prefix.length)
+          .trim();
```

Comment:
I'm not a fan of the whole reformatting. Cutting a 65 char statement into 4 lines makes it way harder to read IMO.

Can you reformat in a different PR so we can isolate these changes?